src/Raven.Client/Documents/Session/EntityToBlittable.cs

```diff
@@ -283,7 +285,26 @@ private static bool TryRemoveIdentityProperty(BlittableJsonReaderObject document
         {
             var identityProperty = conventions.GetIdentityProperty(entityType);
             if (identityProperty == null)
+            {
+                if (conventions.AddIdFieldToDynamicObjects && entityType.GetInterfaces().Contains(typeof(IDynamicMetaObjectProvider)))
+                {
+                    try
+                    {
+                        if (document.Modifications == null)
+                            document.Modifications = new DynamicJsonValue(document);
+
+                        document.Modifications.Remove("Id");
+                        return true;
+                    }
+                    catch (RuntimeBinderException)
```

Comment:
No need to catch RuntimeBinderException. We are not performing any operations on 'dynamic'

Reply:
done